### PR TITLE
Use tooltip [base: refactor-action-2]

### LINKED ISSUE #2594: Add tooltip to invite control
Any iconic controls without a label should have a tooltip, esp. the important ones!

invite tooltip

### Acceptance Criteria
- Tooltip shows on hover
- Tooltip goes away on tap
- Invite modal shows on tap
- Extra points for any reusable pattern for other controls (I know this particular case is tricky because the control is a toggle for a modal portal, and the tooltip is also a portal).

### Estimated Effort
- 10.5 points ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

### PR DESCRIPTION
**Pull request by ackernaut**
### Notes
- fixes #2594 
- adds `useTooltip` pattern similar to `useMenu` and `useModal`
- updates “Invite to Team” control to `useTooltip` and `useModal`
- doesn’t solve for any delay when mousing over and off quickly
- doesn’t solve for tab to focus/blur, but probably should do that eventually

### Tests
- [ ] Hover over Invite to Team control (add icon button by avatars) and tooltip scales in over `150ms`
- [ ] Stop hovering over Invite to Team control and tooltip fades out over `75ms`
- [ ] Invite to Team modal opens when tapping Invite to Team control
- [ ] Invite to Team modal has error state when trying to invite somebody on the team
- [ ] Invite to Team modal closes on success

### CODE REVIEW

**Comment by ackernaut** [comment]
@mattkrick this is ready for review. A few things that I’m not sure about:
- Seems like `useTooltip` and `useMenu` are very redundant and one `useFloatyThing` could handle both, basically I copied `useMenu`, `useMenuPortal`, and `MenuBackground` to be the Tooltip copycat
- Not sure that `useTooltipPortal` needs an ErrorBoundary or Suspense but I left it in there

**Comment by mattkrick** [comment]
> Seems like useTooltip and useMenu are very redundant

This is good! Let's keep them separate for now, we can abstract the pattern once we find a 3rd similar component

> Not sure that useTooltipPortal needs an ErrorBoundary or Suspense but I left it in there

We should remove all loading & async stuff from the tooltip since tooltips should always load sync


Before merging I'd suggest the following:
- Add a delay for showing the tooptip (500ms-ish)
- While hovered the + button with the tooltip showing, click. The tooltip closes, opens, and closes again. Let's eliminate that flicker
- `TOOLTIP_OPEN_MAX` probably doesn't need to be used. For menus, I use that as a way to tell the parent that the children have finished animating in. Since you're using a scale animation, the children load with the parent, so it's not necessary
- loadingDelay, loadingRef, loadingWidth, Suspense and lazy are not used with tooltips & can all be nuked

**Comment by ackernaut** [comment]
Thanks @mattkrick I’ll take a look at those bullets before merging

**Comment by ackernaut** [comment]
- [ ] Add a delay for showing the tooptip (500ms-ish)
- [x] While hovered the + button with the tooltip showing, click. The tooltip closes, opens, and closes again. Let's eliminate that flicker
- [ ] `TOOLTIP_OPEN_MAX` probably doesn't need to be used. For menus, I use that as a way to tell the parent that the children have finished animating in. Since you're using a scale animation, the children load with the parent, so it's not necessary
- [x] loadingDelay, loadingRef, loadingWidth, Suspense and lazy are not used with tooltips & can all be nuked

File: src/universal/types/constEnums.d.ts [hunk 1]
```diff
@@ -41,3 +44,10 @@ export const enum Layout {
   LAYOUT_GRID = 8, // 1x
   ROW_GUTTER = 16 // 2x
 }
+
+// powers of 2 for cheap sense of scale
+export const enum Radius {
```
Comment (mattkrick) [inline src/universal/types/constEnums.d.ts]: if we go the const enum route for border radii i'd opt for having a number in the name Radius.x2, Radius.x4, etc. There may come a day when we have a 6px border radius

**Comment by ackernaut** [comment]
Quick logic for a default tooltip delay:
- When mouse entering, start 500ms timeout, then show
- If exiting before 500ms, cancel tooltip
Questions/notes:
- Is this the default behavior, can be overridden, or do we compose per component depending on UX
- Being inspired by MD as a default starting point:
  - I don’t think they delay (not that we don’t have a better idea!)
  - They typically put a tooltip on all controls that don’t have a text label, even if the same components are repeated multiple times in the view. We’ve pondered only putting them ok key actions that appear once, and not on repetitive, secondary actions (not that the use of those icons is any more clear just b/c they’re repeated)
Any thoughts here @mattkrick ? I’m only going to give myself a few cycles to try to figure this out as the next step, otherwise opt for help or defer.

**Comment by mattkrick** [comment]
> Is this the default behavior, can be overridden, or do we compose per component depending on UX

I think every tooltip should look the same. for now, i can't think of a reason why they wouldn't. if that reason comes along, we can abstract then.

> I don’t think they delay (not that we don’t have a better idea!)

google docs has ~500ms time for tooltips (hover over the share button)

> They typically put a tooltip on all controls that don’t have a text label, even if the same components are repeated multiple times in the view. 

if the tooltip delay is long enough, i'd say let's put em where ever. i think the goal is to try to avoid showing tooltips when the user doesn't wanna see em. if we put tooltips on eg task cards, then we've basically made the viewport a minefield where folks have to be careful where they rest their cursors.

**Comment by ackernaut** [comment]
@mattkrick that sounds good to me re: delay — I’m having a hard time figuring out where to hook in that delay so I’ve paused on this to work on other tasks. Since `useTooltip` is used in 1 place, I’d be okay merging and creating an issue to add delay. Otherwise I may need some direction. Also, I think we should have the option to trigger the tooltip without delay if we still have cases like “Meeting link copied!”.

**Comment by mattkrick** [comment]
fix added to use-tooltip-delay

**Pull request merged**